@bsunanda   There is some error message from StaticAnalyzers as below, although it is not directly from this PR, would you please have a look?

https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c5beef/22527/llvm-analysis/runStaticChecks.log

/pool/condor/dir_11114/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-02-19-1100/src/SimG4CMS/HGCalTestBeam/plugins/HGCalTBCheckGunPosition.cc:99:65: error: expected ';' after expression
    edm::LogVerbatim("HGCSim") << "isHandle valid: " << isHandle valid;
                                                                ^
                                                                ;
/pool/condor/dir_11114/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-02-19-1100/src/SimG4CMS/HGCalTestBeam/plugins/HGCalTBCheckGunPosition.cc:99:57: error: use of undeclared identifier 'isHandle'; did you mean 'isnanl'?
    edm::LogVerbatim("HGCSim") << "isHandle valid: " << isHandle valid;


/pool/condor/dir_11114/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-02-19-1100/src/SimG4CMS/HGCalTestBeam/plugins/HGCalTBCheckGunPosition.cc:99:66: error: use of undeclared identifier 'valid'
    edm::LogVerbatim("HGCSim") << "isHandle valid: " << isHandle valid;
                                                                 ^
/pool/condor/dir_11114/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-02-19-1100/src/SimG4CMS/HGCalTestBeam/plugins/HGCalTBCheckGunPosition.cc:114:7: error: use of undeclared identifier 'z'
      z = (*p)->position().z() / 10.;  // in cm
      ^
/pool/condor/dir_11114/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-02-19-1100/src/SimG4CMS/HGCalTestBeam/plugins/HGCalTBCheckGunPosition.cc:117:92: error: use of undeclared identifier 'z'
                                   << ":" << y << " z: " << (*p)->position().z() << ":" << z;
                                                                                           ^
5 errors generated.

--- a/SimG4CMS/HGCalTestBeam/plugins/HGCalTBCheckGunPosition.cc
+++ b/SimG4CMS/HGCalTestBeam/plugins/HGCalTBCheckGunPosition.cc
@@ -96,7 +96,7 @@ bool HGCalTBCheckGunPostion::filter(edm::Event& iEvent, const edm::EventSetup& i
   const edm::Handle<edm::HepMCProduct>& hepmc = iEvent.getHandle(hepMCproductLabel_);
 #ifdef EDM_ML_DEBUG
   if (verbosity_)
-    edm::LogVerbatim("HGCSim") << "isHandle valid: " << isHandle valid;
+    edm::LogVerbatim("HGCSim") << "is hepmc Handle valid: " << hepmc.isValid();
 #endif
   double x(0), y(0);
 
@@ -111,7 +111,7 @@ bool HGCalTBCheckGunPostion::filter(edm::Event& iEvent, const edm::EventSetup& i
       x = (*p)->position().x() / 10.;  // in cm
       y = (*p)->position().y() / 10.;  // in cm
 #ifdef EDM_ML_DEBUG
-      z = (*p)->position().z() / 10.;  // in cm
+      double z = (*p)->position().z() / 10.;  // in cm
       if (verbosity_)
         edm::LogVerbatim("HGCSim") << " x: " << (*p)->position().x() << ":" << x << " y: " << (*p)->position().y()
                                    << ":" << y << " z: " << (*p)->position().z() << ":" << z;

